### PR TITLE
Search controller refactoring

### PR DESCRIPTION
**Pull request by vpereira**
Extract code from `SearchController#find_attribute` to methods

There are more refactoring to come, for now, just some cleaning tasks.

-----------------

If this PR requires any particular action or consideration before deployment,
please check the reasons or add your own to the list:

* [ ] Requires a database migration that can cause downtime. Postpone deployment until maintenance window[1].
* [ ] Contains a data migration that can cause temporary inconsistency, so should be run at a specific point of time.
* [ ] Changes some configuration files (e.g. options.yml), so the changes have to be applied manually in the reference server.
* [ ] A new Feature Toggle[2] should be enabled in the reference server.
* [ ] Proper documentation or announcement has to be published upfront since the introduced changes can confuse the users.

[1] https://github.com/openSUSE/open-build-service/wiki/Deployment-of-build.opensuse.org#when-there-are-migrations
[2] https://github.com/openSUSE/open-build-service/wiki/Feature-Toggles-%28Flipper%29#you-want-real-people-to-test-your-feature

### CODE REVIEW

**Comment by hennevogel** [comment]
Will you move over the rest of the tests from `test/functional/search_controller_test.rb` too?

File: src/api/spec/controllers/search_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe SearchController, vcr: true do
+  render_views
+
+  shared_examples 'block anonymous request' do
+    it { expect(response).to have_http_status(:redirect) }
```
Comment (hennevogel) [inline src/api/spec/controllers/search_controller_spec.rb:8]: I don't think this makes much sense. `require_login` is a "default" `before_filter` from `ApplicationController`.

**Comment by vpereira** [comment]
> Will you move over the rest of the tests from `test/functional/search_controller_test.rb` too?

Yes, we should at least cover all the scenarios covered by minitest, yes :+1:

File: src/api/spec/controllers/search_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  end
+
+  shared_examples 'find project' do
+    it { expect(response).to have_http_status(:success) }
+    it { expect(Xmlhash.parse(response.body)).to have_key('project') }
+    it { expect(Xmlhash.parse(response.body)['project']).to have_key('name') }
+    it { expect(Xmlhash.parse(response.body)['project']['name']).to eq(user.home_project.name) }
```
Comment (hennevogel) [inline src/api/spec/controllers/search_controller_spec.rb:15]: Aren't those 3 the same test? If you don't have a key project you will for sure not have `['project']['name']` with some expected content.

File: src/api/spec/controllers/search_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  end
+
+  shared_examples 'find package' do
+    it { expect(response).to have_http_status(:success) }
+    it { expect(Xmlhash.parse(response.body)).to have_key('package') }
+    it { expect(Xmlhash.parse(response.body)['package']).to have_key('name') }
+    it { expect(Xmlhash.parse(response.body)['package']['name']).to eq('apacheX') }
```
Comment (hennevogel) [inline src/api/spec/controllers/search_controller_spec.rb:22]: Same here...

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/spec/controllers/search_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+      it_behaves_like 'block anonymous request'
+    end
+
+    context 'authenticated' do
+      let!(:user) { create(:confirmed_user, :with_home, login: 'foo') }
+
+      before do
+        login user
+      end
```
Comment (hennevogel) [inline src/api/spec/controllers/search_controller_spec.rb:38]: You can move this into the shared example.

Comment (vpereira) [reply src/api/spec/controllers/search_controller_spec.rb]: done

File: src/api/spec/controllers/search_controller_spec.rb (hunk 1 above)
Comment (vpereira) [reply src/api/spec/controllers/search_controller_spec.rb]: done

File: src/api/spec/controllers/search_controller_spec.rb (hunk 2 above)
Comment (vpereira) [reply src/api/spec/controllers/search_controller_spec.rb:17]: your suggestion is to keep just the last one?

Comment (hennevogel) [reply src/api/spec/controllers/search_controller_spec.rb:17]: jupp

File: src/api/spec/controllers/search_controller_spec.rb (hunk 3 above)
Comment (vpereira) [reply src/api/spec/controllers/search_controller_spec.rb]: done

File: src/api/spec/controllers/search_controller_spec.rb (hunk 2 above)
Comment (vpereira) [reply src/api/spec/controllers/search_controller_spec.rb]: done

File: src/api/spec/controllers/search_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe SearchController, vcr: true do
+  render_views
+
+  let!(:user) { create(:confirmed_user, :with_home, login: 'foo') }
```
Comment (hennevogel) [inline src/api/spec/controllers/search_controller_spec.rb:7]: nitpick: `let` should be enough...

**Review by hennevogel — APPROVED** [review]

**Pull request merged**